contract_sale_generation/tests/test_contract_sale.py

```diff
@@ -96,6 +96,11 @@ def test_cron_recurring_create_sale(self):
         self.contract_line.recurring_invoicing_type = "post-paid"
         self.contract_line.date_end = "2020-03-15"
         self.contract_line._onchange_is_auto_renew()
+        # If we do not recompute recurring_next_date
```

Comment:
Yes, IMHO there are still some cases that should be solved after theses changes : https://github.com/OCA/contract/commit/cd086ddbb4a9b85ccfcac288e53fb93e716d9446